newton/_src/viewer/viewer_viser.py

```
@@ -190,33 +196,167 @@ def _call_scene_method(method, **kwargs):
+
+        Returns:
+            str: Local HTTP URL for the running viser server.
+        """
         return f"http://localhost:{self._port}"
 
+    @staticmethod
+    def _is_client_camera_ready(client: Any) -> bool:
+        """Return True if the client has reported an initial camera state."""
+        try:
+            _ = client.camera.position
+        except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)